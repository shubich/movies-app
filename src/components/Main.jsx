import React from 'react';
import PropTypes from 'prop-types';
import ReactLoading from 'react-loading';
import Film from './Film';
import Person from './Person';
import NotFound from './NotFound';

const Main = props => (
  (props.results.length || props.fetching) ?
    <main id="main">
      <div className="container">
        {
        props.fetching ?
          <ReactLoading type="bars" color="#000" delay={0} />
        :
        props.results.map((item) => {
          if (props.searchType === 'title') {
            return <Film key={item.id} film={item} />;
          } else if (props.searchType === 'person') {
            return <Person key={item.id} person={item} />;
          }
          return null;
        })
      }
      </div>
    </main>
    :
    <NotFound />
);

Main.propTypes = {
  results: PropTypes.arrayOf(PropTypes.object).isRequired,
  fetching: PropTypes.bool.isRequired,
  searchType: PropTypes.string.isRequired,
};

export default Main;
