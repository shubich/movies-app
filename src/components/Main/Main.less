@import "../../styles/palette";

.Main {
    background: @default-color;
    min-height: 550px;
}