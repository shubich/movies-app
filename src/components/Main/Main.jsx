import React from 'react';
import PropTypes from 'prop-types';

import './Main.less';

const Main = props => (
  <main className="Main">
    {props.children}
  </main>
);

Main.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.array,
  ]).isRequired,
};

export default Main;
