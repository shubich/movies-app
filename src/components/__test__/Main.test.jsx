import React from 'react';
import { Main } from '../Main';
import { shallow } from 'enzyme';

let props;

describe('Main', () => {
  beforeEach(() => {
    props = {
      results: [
        {
          id: 1,
          title: 'A',
        },
        {
          id: 2,
          title: 'B',
        },
      ],
      fetching: false,
      error: '',
      onFilmClick: jest.fn,
      searchType: 'title',
    };
  });

  it('renders correctly with props', () => {
    const wrapper = shallow(<Main {...props} />);
    expect(wrapper).toMatchSnapshot();
  });


  it('renders correctly with fetching', () => {
    const wrapper = shallow(<Main {...props} fetching />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly without results', () => {
    const wrapper = shallow(<Main {...props} results={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly without searchType', () => {
    const wrapper = shallow(<Main {...props} searchType="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
