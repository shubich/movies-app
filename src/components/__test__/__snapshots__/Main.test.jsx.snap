// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly with fetching 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    error=""
    fetching={true}
    onFilmClick={[Function]}
    results={
      Array [
        Object {
          "id": 1,
          "title": "A",
        },
        Object {
          "id": 2,
          "title": "B",
        },
      ]
    }
    searchType="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <Loading
          className=""
          color="#000"
          delay={0}
          height={64}
          type="bars"
          width={64}
        />
      </div>,
      "id": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Loading
          className=""
          color="#000"
          delay={0}
          height={64}
          type="bars"
          width={64}
        />,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "className": "",
          "color": "#000",
          "delay": 0,
          "height": 64,
          "type": "bars",
          "width": 64,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <Loading
            className=""
            color="#000"
            delay={0}
            height={64}
            type="bars"
            width={64}
          />
        </div>,
        "id": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <Loading
            className=""
            color="#000"
            delay={0}
            height={64}
            type="bars"
            width={64}
          />,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "color": "#000",
            "delay": 0,
            "height": 64,
            "type": "bars",
            "width": 64,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Main renders correctly with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    error=""
    fetching={false}
    onFilmClick={[Function]}
    results={
      Array [
        Object {
          "id": 1,
          "title": "A",
        },
        Object {
          "id": 2,
          "title": "B",
        },
      ]
    }
    searchType="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <Film
          film={
            Object {
              "id": 1,
              "title": "A",
            }
          }
        />
        <Film
          film={
            Object {
              "id": 2,
              "title": "B",
            }
          }
        />
      </div>,
      "id": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Film
            film={
              Object {
                "id": 1,
                "title": "A",
              }
            }
          />,
          <Film
            film={
              Object {
                "id": 2,
                "title": "B",
              }
            }
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "film": Object {
              "id": 1,
              "title": "A",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "film": Object {
              "id": 2,
              "title": "B",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <Film
            film={
              Object {
                "id": 1,
                "title": "A",
              }
            }
          />
          <Film
            film={
              Object {
                "id": 2,
                "title": "B",
              }
            }
          />
        </div>,
        "id": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Film
              film={
                Object {
                  "id": 1,
                  "title": "A",
                }
              }
            />,
            <Film
              film={
                Object {
                  "id": 2,
                  "title": "B",
                }
              }
            />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "film": Object {
                "id": 1,
                "title": "A",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "film": Object {
                "id": 2,
                "title": "B",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Main renders correctly without results 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    error=""
    fetching={false}
    onFilmClick={[Function]}
    results={Array []}
    searchType="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Main renders correctly without searchType 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    error=""
    fetching={false}
    onFilmClick={[Function]}
    results={
      Array [
        Object {
          "id": 1,
          "title": "A",
        },
        Object {
          "id": 2,
          "title": "B",
        },
      ]
    }
    searchType=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      />,
      "id": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
      ],
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        />,
        "id": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            null,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          null,
          null,
        ],
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
